### PR TITLE
refactor spring messaging storage queue

### LINKED ISSUE #26273: Refactor spring-cloud-azure-messaging-storage-queue to follow the same convention of spring-cloud-azure-messaging-* libraries
Refactor the code structure of **spring-cloud-azure-messaging-storage-queue** library to keep the same convention with other messaging libraries. Changes should include:
- Remove checkpoint logic
- Remove StorageQueueOperation
- Add Factory abstraction to create SDK client
- Add properties abstraction
Ref: https://gist.github.com/yiliuTo/352b071a717fd90c984bcf03ec5d0fb9

### PR DESCRIPTION
**Pull request by yiliuTo**
resolve #26273

### CODE REVIEW

File: sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/factory/StorageQueueClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// Licensed under the MIT License.
+
+package com.azure.spring.storage.queue.core.factory;
+
+import com.azure.storage.queue.QueueAsyncClient;
+
+/**
+ * The strategy to produce {@link QueueAsyncClient} instance.
+ */
+public interface StorageQueueClientFactory {
```
Comment (saragluna) [inline sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/factory/StorageQueueClientFactory.java:11]: Should we unify this class name for these factories? with event hubs and service bus?

File: sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/properties/StorageQueueProperties.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+import com.azure.spring.core.properties.AzureHttpSdkProperties;
+import com.azure.spring.service.storage.queue.QueueServiceClientProperties;
+import com.azure.storage.queue.QueueMessageEncoding;
+import com.azure.storage.queue.QueueServiceVersion;
+
+/**
+ * Properties for Azure Storage Queue service.
+ */
+public class StorageQueueProperties extends AzureHttpSdkProperties implements QueueServiceClientProperties {
```
Comment (saragluna) [inline sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/properties/StorageQueueProperties.java:13]: Do we already have one in the service module?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-integration-azure-storage-queue/CHANGELOG.md [hunk 3]
```diff
@@ -3,9 +3,12 @@
 ## 4.0.0-beta.3 (Unreleased)
 
 ### Features Added
-
+- Support StorageQueueMessageConverter as a bean to support customize ObjectMapper.
 ### Breaking Changes
-
+- Remove `StorageQueueOperation`.
+- Deprecate configure checkpoint mode for StorageQueueTemplate, and support only MANUAL mode.
+- Remove auto creating Storage Queue when send/receive messages via `StorageQueueTemplate`, and provide a property of
```
[the remark this answers is not in the document]
Comment (yiliuTo) [reply sdk/spring/spring-integration-azure-storage-queue/CHANGELOG.md:10]: Forgot to update the changelog, will modify it

File: sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/StorageQueueTemplate.java [hunk 4]
```diff
@@ -66,30 +70,17 @@ public StorageQueueTemplate(@NonNull StorageQueueClientFactory storageQueueClien
         return this.receiveAsync(queueName, visibilityTimeout);
     }
 
-    private Mono<Message<?>> receiveAsync(String queueName, Duration visibilityTimeout) {
+    public Mono<Message<?>> receiveAsync(String queueName, Duration visibilityTimeout) {
```
[the remark this answers is not in the document]
Comment (yiliuTo) [reply sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/StorageQueueTemplate.java:73]: We could reserve this api to follow SDK's behavior that provide a default visibilityTimeout value.

File: sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/properties/StorageQueueProperties.java (hunk 2 above)
Comment (yiliuTo) [reply sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/properties/StorageQueueProperties.java:13]: we have an interface of `QueueServiceClientProperties` in service module but not a POJO class.

File: sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/StorageQueueTemplate.java [hunk 5]
```diff
@@ -169,25 +134,35 @@ public void setVisibilityTimeout(Duration visibilityTimeoutDuration) {
...
+     */
+    @Deprecated
     public void setCheckpointMode(CheckpointMode checkpointMode) {
-        Assert.state(isValidCheckpointMode(checkpointMode),
-            "Only MANUAL or RECORD checkpoint mode is supported in StorageQueueTemplate");
-        this.checkpointMode = checkpointMode;
-        LOG.info("StorageQueueTemplate checkpoint mode becomes: {}", this.checkpointMode);
+        throw new IllegalStateException("Configuration of checkpoint mode is not supported, the MANUAL checkpoint mode is applied");
     }
+
```
Comment (saragluna) [inline sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/StorageQueueTemplate.java:167]: We should delete them.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**